File: leetcode/lc-us/cpp/1658.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int minOperations(vector<int> &nums, int x) {
        int size = (int) nums.size(), INF = 1e9;
        for (int i = 0; i < size; i++) nums.push_back(nums[i]);
        int n = (int) nums.size();
        vector<int> s(n + 1);
        for (int i = 1; i <= n; i++) s[i] = s[i - 1] + nums[i - 1];
        int res = INF;
        for (int i = 1; i <= size; i++) {
            int l = i, r = i + size - 1;
            while (l < r) {
                int mid = (l + r) / 2;
                int sum = s[mid] - s[i - 1];
                if (sum < x) l = mid + 1;
                else r = mid;
            }
            if (s[l] - s[i - 1] == x && (l >= size || i == 1)) {
                res = min(res, l - i + 1);
            }
        }
        return res == INF ? -1 : res;
    }
};

File: leetcode/lc-us/py/1658.py
from typing import List


class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        n = len(nums)
        res = n + 1
        s, i, j = 0, 0, 0
        while i < n:
            while j < n + i + 1 and s < x:
                s += nums[j % n]
                j += 1
            if s == x:
                if i == 0 or j >= n:
                    res = min(res, j - i)
                s -= nums[i]
                i += 1
            elif s > x:
                while i < n and s > x:
                    s -= nums[i]
                    i += 1
            else:
                break
        return -1 if res == n + 1 else res
